src/DynamoCore/Nodes/NodeCategories.cs

```diff
@@ -454,6 +454,147 @@ internal static string InsertSpacesToString(string original)
+
+                if (Char.IsUpper(curr) && i != 0 && !Char.IsUpper(prev) && (!curr.Equals(" ")))
+                {
+                    words.Add(currentWord.ToString());
+                    currentWord.Clear();
+                }
+                currentWord.Append(curr);
+            }
+            // Add last word.
+            words.Add(currentWord.ToString());
+            return words;
+        }
+
+        /// <summary>
+        /// Merge rows according max nuber characters per row, e.g. ("Day","Of","Week") => ("Day Of","Week")
+        /// Last row can contain remaining rows, if there is no place to store them,
+        /// e.g. ("Import","From","CSV") => ("Import","From CSV")
+        /// </summary>
+        /// <param name="rows">Incoming rows</param>
+        /// <param name="maxRows">Max number of rows</param>
+        /// <param name="maxCharacters">Max number characters per row</param>
+        internal static IEnumerable<string> ReduceRowCount(IEnumerable<string> rows, int maxRows, int maxCharacters)
+        {
+            if (rows == null || maxRows <= 0 || maxCharacters <= 0)
+                throw new ArgumentException();
+            if (rows.Count() == 0 || rows.Count() == maxRows)
+                return rows;
+            List<string> resultRows = new List<string>();
+
+            StringBuilder currentRow = new StringBuilder();
+            foreach (var row in rows)
+            {
+                // +1, because we need to add space between rows.
+                if ((currentRow.Length + row.Length + 1) <= maxCharacters)
+                {
+                    if (currentRow.Length != 0)
+                        currentRow.Append(" ");
+                    currentRow.Append(row);
+                }
+                else
+                {
+                    if (resultRows.Count < maxRows - 1)
+                    {
+                        resultRows.Add(currentRow.ToString());
+                        currentRow.Clear();
+                        currentRow.Append(row);
+                    }
+                    else
+                    {
+                        if (currentRow.Length != 0)
+                            currentRow.Append(" ");
+                        currentRow.Append(row);
+                    }
+                }
+            }
+            // Add last row, if it's not empty.
+            if (currentRow.Length != 0)
+                resultRows.Add(currentRow.ToString());
+
+            return resultRows;
+        }
+
+        /// <summary>
+        /// Trincate rows by adding dots, e.g. ("Surface", "Analysis Data") => ("Surface..","..Data")
+        /// </summary>
+        /// <param name="rows">Incomming rows</param>
+        /// <param name="maxCharacters">Max number characters per row</param>
+        internal static IEnumerable<string> TruncateRows(IEnumerable<string> rows, int maxCharacters)
+        {
+            if (rows == null || maxCharacters <= 0)
```

Comment:
I think `maxCharacter` cannot be less than or equals to `2`, because we need to append two dots, so it must be more than `2` :)